python/dask_cudf/dask_cudf/backends.py

```diff
@@ -10,10 +13,78 @@
 get_parallel_type.register(cudf.Index, lambda _: Index)
 
 
-@meta_nonempty.register((cudf.DataFrame, cudf.Series, cudf.Index))
-def meta_nonempty_cudf(x, index=None):
-    y = meta_nonempty(x.to_pandas())  # TODO: add iloc[:5]
-    return cudf.from_pandas(y)
+@meta_nonempty.register(cudf.Index)
+def _nonempty_index(idx):
+    if isinstance(idx, cudf.core.index.RangeIndex):
+        return cudf.core.index.RangeIndex(2, name=idx.name)
+    elif isinstance(idx, cudf.core.index.DatetimeIndex):
+        start = "1970-01-01"
+        data = [start, "1970-01-02"]
+        return cudf.core.index.DatetimeIndex(
+            data, start=start, periods=2, name=idx.name
+        )
```

Comment:
How can we handling different dtype resolutions within this? I.E. `ns` vs `ms` vs `s`?

Reply:
Oh no - Am I going to need to learn more about datetime dtypes? :)

For pandas, dask passes more information(like timezone and frequency) to the `pd.DatetimeIndex` initializer.  I'll see what we can do for cudf to capture the precision.

Reply:
For context: Pandas always uses `ns` resolution but we allow specifying the resolution, so we'll need to differ from Pandas here.

Reply:
Thanks @kkraus14 - Different dtype resolutions should now be covered here